File: notify-service-send-email/src/main/java/com/hoangtien2k3/notifyservicesendemail/service/EmailService.java
package com.hoangtien2k3.notifyservicesendemail.service;

import com.hoangtien2k3.notifyservicesendemail.entity.EmailDetails;
import org.springframework.web.multipart.MultipartFile;

public interface EmailService {
    // To send a simple email
    String sendSimpleMail(EmailDetails details);

    // To send an email with attachment
    String sendMailWithAttachment(EmailDetails details);

    // send email file
    String sendMail(MultipartFile[] file, String to, String[] cc, String subject, String body);
}

File: send-email-service/src/main/kotlin/com/hoangtien2k3/sendemailservice/service/EmailService.kt
package com.hoangtien2k3.sendemailservice.service

import com.hoangtien2k3.sendemailservice.entity.EmailDetails
import org.springframework.web.multipart.MultipartFile

interface EmailService {
    fun sendSimpleMail(details: EmailDetails?): String?

    fun sendMailWithAttachment(details: EmailDetails?): String?

    fun sendMail(
        file: Array<MultipartFile?>?,
        to: String?,
        cc: Array<String?>?,
        subject: String?,
        body: String?
    ): String?

}